proxy-node-gateway/src/test/java/uk/gov/ida/notification/apprule/rules/TestSamlResponseExceptionMapperRule.java

```diff
@@ -0,0 +1,108 @@
+        when(samlFormViewBuilder.buildResponse("EidasDestination", nullString, "Continue", "EidasRelayState"))
+                .thenReturn(
+                new SamlFormView("postUrl",
+                                 "samlMessageType",
+                                 "encodedSamlMessage",
+                                 "submitText"));
+    }
+
+    @ClassRule
+    public static final ResourceTestRule resources = ResourceTestRule.builder()
+                                    .setTestContainerFactory(new GrizzlyWebTestContainerFactory())
+                                    .addProvider(new ExceptionToSamlErrorResponseMapper(samlFormViewBuilder, translatorProxy, sessionStore))
+                                    .addResource(new TestExceptionMapperResource())
+                                    .build();
+
+    @Test
+    public void shouldMapSessionAlreadyExistsExceptionToExceptionToSamlErrorResponseMapper() {
+
+        var response = resources.getJerseyTest()
+                                .target("SessionAlreadyExistsException")
+                                .request()
+                                .get();
+        assertThat(response.getStatusInfo()).isEqualTo(Response.Status.OK);
+        assertThat(checkResponseEntityIsSamlFormResponse(response.readEntity(String.class))).isTrue();
+    }
+
+    @Test
+    public void shouldMapSessionAttributeExceptionToExceptionToSamlErrorResponseMapper() {
+        var response = resources.getJerseyTest()
+                                .target("/SessionAttributeException")
+                                .request()
+                                .get();
+        assertThat(response.getStatusInfo()).isEqualTo(Response.Status.OK);
+        assertThat(checkResponseEntityIsSamlFormResponse(response.readEntity(String.class))).isTrue();
+    }
+
+    @Test
+    public void shouldMapTranslatorResponseExceptionToExceptionToSamlErrorResponseMapper() {
+        var response = resources.getJerseyTest()
+                                .target("/TranslatorResponseException")
+                                .request()
+                                .get();
+        assertThat(response.getStatusInfo()).isEqualTo(Response.Status.OK);
+        assertThat(checkResponseEntityIsSamlFormResponse(response.readEntity(String.class))).isTrue();
+    }
+
+    private boolean checkResponseEntityIsSamlFormResponse(String responseEntity) {
+        // The actual application returns a web page with a form but this stripped down version, returns some JSON
+        // As long as it matches, we can be confident that the correct exception mapper has been used
+        Pattern pattern = Pattern.compile("\\{\"postUrl\":\"postUrl\",\"samlMessageType\":\"samlMessageType\",\"encodedSamlMessage\":\"encodedSamlMessage\",\"submitText\":\"submitText\",\"relayState\":.*\"}");
+        Matcher matcher = pattern.matcher(responseEntity);
+        return matcher.matches();
+    }
+
+    private static URI getUri() {
```

Comment:
Is this method used anywhere?

Reply:
No, it's a left over from when both mappers were being tested in the same class.  It has now been deleted.